### DISCUSSION TITLE
Why is the template generate a CMakeUserPresets, should not those be outside of git and local to the dev own setup?

### DISCUSSION THREAD

**Discussion by hpe-ykoehler** [friendlyanon/cmake-init#83] [Q&A]
Still learning about cmake, saw the presets stuff, found this project.  Tried it, and I see a CMakeUserPreset generated, but isn't this file suppose to be for local only? 

I would have expected all the preset in there to be part of the CMakePresets and not have a CMakeUserPresets unless I create one myself.

https://cmake.org/cmake/help/latest/manual/cmake-presets.7.html
```
CMakePresets.json is meant to specify project-wide build details, while CMakeUserPresets.json is meant for developers to specify their own local build details.

CMakePresets.json may be checked into a version control system, and CMakeUserPresets.json should NOT be checked in
```

**Comment by friendlyanon** [comment]
The user presets are in fact outside of git:

https://github.com/friendlyanon/cmake-init/blob/5cb71c90489dedf10967305be2d5a040d5d976ce/cmake-init/templates/common/.gitignore#L10

The developer guide also says that user presets where all the local settings could comfortably go:

https://github.com/friendlyanon/cmake-init/blob/5cb71c90489dedf10967305be2d5a040d5d976ce/cmake-init/templates/common/HACKING.md#L74-L76

**Comment by hpe-ykoehler** [comment]
Yes, but my understanding is that this is the place where a developer can customize for his own benefit, but in cmake-init the dev-* presets are put in this file.  That dev-* presets looks more like a project-wide dev setting than a local one.

Basically, only the developer should play with his own CMakeUserPresets, and he should basically be the one creating the file on his local PC, if I understood this properly.

**Reply by friendlyanon** [reply]
The file is created to help the developer get started easily. The contributing and hacking documents contain information to recreate the file for the developer and contributors. There is no conflict here. You are free to remove the generated file for yourself and roll your own, but I think it helps greatly when getting started. The GIF in the readme is the experience I aim for, that is to have a working project right away.
